Library/Homebrew/cmd/uninstall.rb

```diff
@@ -73,5 +78,9 @@ def uninstall
         force:   args.force?,
       )
     end
+
+    return unless Homebrew::EnvConfig.uninstall_autoremove?
+
+    system HOMEBREW_BREW_FILE, "autoremove"
```

Comment:
Sure, I can do that. But now I'm curious, is there some sort of general rule when deciding whether to shell out to `brew` or is that sort of a feel thing?

Reply:
Probably a function of how often the code will be exercised. This might not be used that often when gated behind an environment variable, but there was a suggestion to make this the default behaviour (in which case it will probably be used very often).

Reply:
> is there some sort of general rule when deciding whether to shell out to `brew` or is that sort of a feel thing?

It probably depends who you ask. I'm more in the camp of "shell out as little as possible" than probably most - Ruby API is generally nicer to deal with.

Reply:
> I'm more in the camp of "shell out as little as possible" than probably most - Ruby API is generally nicer to deal with.

Same. Shelling out is always slower. The only real advantage is that it can be "cleaner" in terms of not preserving internal Ruby state but that's something I think is worth generally fixing.